### PR TITLE
Add MSBuild props for run-time config

### LINKED ISSUE #16742: Document setting config options in the .csproj file
Document [run-time config settings](https://github.com/dotnet/sdk/blob/a5f5bb5183517cb301050288b25c9c727e52e8a6/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L344-L380) that can be specified in the project file. Add them to [this section of the docs](https://docs.microsoft.com/en-us/dotnet/core/run-time-config/) as well as the [new article](#2642) on msbuild properties that are known to the .NET Core SDK.

### PR DESCRIPTION
**Pull request by gewarren**
- add MSBuild properties where available
- flesh out examples for JSON and MSBuild props
- explain the runtimeconfig.template.json and appname.runtimeconfig.json relationship
- update tiered compilation explanations

[Internal preview link for compilation](https://review.docs.microsoft.com/en-us/dotnet/core/run-time-config/compilation?branch=pr-en-us-16794)

Fixes #16742

### CODE REVIEW

File: docs/core/run-time-config/compilation.md [hunk 1]
```diff
@@ -8,11 +8,112 @@ ms.topic: reference
 
 ## Tiered compilation
 
-- Configures whether the JIT compiler uses [tiered compilation](../whats-new/dotnet-core-3-0.md#tiered-compilation).
+- Configures whether the just-in-time (JIT) compiler uses [tiered compilation](../whats-new/dotnet-core-3-0.md#tiered-compilation).
 - In .NET Core 3.0 and later, tiered compilation is enabled by default.
 - In .NET Core 2.1 and 2.2, tiered compilation is disabled by default.
+- For more information, see the [Tiered compilation](https://devblogs.microsoft.com/dotnet/tiered-compilation-preview-in-net-core-2-1/) blog entry.
```
Comment (kouvel) [inline docs/core/run-time-config/compilation.md]: This blog post is a bit dated. The current guide for TC is at https://github.com/dotnet/runtime/blob/master/docs/design/features/tiered-compilation-guide.md. It is also a bit dated but that's the one that'll be kept updated (hopefully soon). The guide references the blog post.

File: docs/core/run-time-config/compilation.md [hunk 2]
```diff
@@ -8,10 +8,10 @@ ms.topic: reference
 
 ## Tiered compilation
 
-- Configures whether the just-in-time (JIT) compiler uses [tiered compilation](../whats-new/dotnet-core-3-0.md#tiered-compilation).
+- Configures whether the just-in-time (JIT) compiler uses [tiered compilation](../whats-new/dotnet-core-3-0.md#tiered-compilation). Tiered compilation includes two distinct behaviors: [quick JIT](#quick-jit) and the use of [ReadyToRun](../whats-new/dotnet-core-3-0.md#readytorun-images) (pre-compiled) code.
```
Comment (kouvel) [inline docs/core/run-time-config/compilation.md]: > Tiered compilation includes two distinct behaviors

Perhaps:
"Tiered compilation transitions methods through two tiers: the first tier generates code more quickly (quick JIT) or loads pre-compiled code (ReadyToRun), and the second tier generates optimized code in the background (optimizing JIT)."

**Review by kouvel — COMMENTED** [review]
Just a suggestion, otherwise tiering and threading stuff look good to me, thanks!

File: docs/core/run-time-config/garbage-collector.md [hunk 3]
```diff
@@ -255,10 +285,13 @@ Example:
 
 | | Setting name | Values | Version introduced |
 | - | - | - | - |
-| **runtimeconfig.json** | `System.GC.RetainVM` | `false` - release to OS<br/>`true` - put on standby| .NET Core 1.0 |
+| **runtimeconfig.json** | `System.GC.RetainVM` | `false` - release to OS<br/>`true` - put on standby | .NET Core 1.0 |
```
Comment (Thraka) [inline docs/core/run-time-config/garbage-collector.md]: nit: What good is the version introduced column here? We could probably delete it.

File: docs/core/tools/dotnet.md [hunk 4]
```diff
@@ -101,7 +101,7 @@ Defines behavior when the required shared framework is not available. `N` can be
 
 Path to a *runtimeconfig.json* file.
 
-A *runtimeconfig.json* file is a configuration file containing runtime configuration settings. For more information, see [Runtime Configuration Files](https://github.com/dotnet/cli/blob/master/Documentation/specs/runtime-configuration-file.md) on GitHub.
+A *runtimeconfig.json* file is a configuration file containing run-time settings. For more information, see [.NET Core run-time configuration settings](../run-time-config/index.md#runtimeconfigjson) on GitHub.
```
Comment (Thraka) [inline docs/core/tools/dotnet.md]: This file is no longer on github but in the docs.

```suggestion
A *runtimeconfig.json* file is a configuration file containing run-time settings. For more information, see [.NET Core run-time configuration settings](../run-time-config/index.md#runtimeconfigjson).
```

File: docs/core/run-time-config/garbage-collector.md (hunk 3 above)
Comment (Thraka) [inline docs/core/run-time-config/garbage-collector.md]: nit: What good is the version introduced column here? We could probably delete it.

File: docs/core/run-time-config/compilation.md [hunk 5]
```diff
@@ -8,11 +8,118 @@ ms.topic: reference
 
 ## Tiered compilation
 
-- Configures whether the JIT compiler uses [tiered compilation](../whats-new/dotnet-core-3-0.md#tiered-compilation).
+- Configures whether the just-in-time (JIT) compiler uses [tiered compilation](../whats-new/dotnet-core-3-0.md#tiered-compilation). Tiered compilation includes two distinct behaviors: [quick JIT](#quick-jit) and the use of [ReadyToRun](../whats-new/dotnet-core-3-0.md#readytorun-images) (pre-compiled) code.
```
Comment (Thraka) [inline docs/core/run-time-config/compilation.md]: Also, are you going to port the ReadyToRun info from the whats new to here?

**Review by Thraka — APPROVED** [review]
LGTM, left you a few notes and fixes.

File: docs/core/run-time-config/compilation.md [hunk 6]
```diff
@@ -8,11 +8,118 @@ ms.topic: reference
...
 - In .NET Core 3.0 and later, tiered compilation is enabled by default.
 - In .NET Core 2.1 and 2.2, tiered compilation is disabled by default.
+- For more information, see the [Tiered compilation guide](https://github.com/dotnet/runtime/blob/master/docs/design/features/tiered-compilation-guide.md).
```
Comment (mairaw) [inline docs/core/run-time-config/compilation.md]: not related to this PR, but this seems like something we should add to our docs, right?

File: docs/core/run-time-config/garbage-collector.md (hunk 3 above)
Comment (gewarren) [reply docs/core/run-time-config/garbage-collector.md]: Maoni requested it, so I think we should leave it.

Comment (gewarren) [reply docs/core/run-time-config/garbage-collector.md]: Some knobs were introduced in later versions of .NET Core.

File: docs/core/run-time-config/compilation.md (hunk 5 above)
Comment (gewarren) [reply docs/core/run-time-config/compilation.md]: I think this goes along with @mairaw Maira's comment about the fact that we need to document tiered compilation conceptually. So I think the R2R info would go into that doc. I'll create an issue for it. Some of the TC detail in this config reference could then be pulled out to that doc too.

File: docs/core/run-time-config/compilation.md (hunk 6 above)
Comment (gewarren) [reply docs/core/run-time-config/compilation.md]: Created https://github.com/dotnet/docs/issues/16805

File: docs/core/run-time-config/compilation.md (hunk 5 above)
Comment (Thraka) [reply docs/core/run-time-config/compilation.md]: I'm just referring to the table structure for the config vs xml vs environment variable setting. It looks so nice in here compared to the whats new 😄

Comment (gewarren) [reply docs/core/run-time-config/compilation.md]: Ah, yes we have issue https://github.com/dotnet/docs/issues/16781, which I'll address separately after this merges to master.

**Pull request merged**